musicbot/bot.py

```diff
@@ -1950,6 +1950,27 @@ def check(m):
                 raise exceptions.CommandError(
                     'Unreasonable volume provided: {}%. Provide a value between 1 and 100.'.format(new_volume), expire_in=20)
 
+    async def cmd_autoplaylist(self, channel, author, player, voice_channel):
+        """
+        Usage:
+            {command_prefix}autoplaylist
+			
+		Enable or disable the autoplaylist
+        """
+        song_url = random.choice(self.autoplaylist)       
+        if self.config.auto_playlist == False:
+            self.config.auto_playlist = True
+            if not player.playlist.entries and not player.current_entry and self.config.auto_playlist: #if nothing is queued, start a song
+                await player.playlist.add_entry(song_url, channel=None, author=None)
+            await self.safe_send_message(channel, "The autoplaylist is now enabled!")
+            return
+			
+        if self.config.auto_playlist == True: #else disable 
+            self.config.auto_playlist = False
+            await self.safe_send_message(channel, "The autoplaylist is now disabled!")
+            return							
+	
```

Comment:
RE: The whole thing, it may just make sense to invert the `self.config.auto_playlist` boolean and then do nothing else. That'll mean that after the current song has finished playing, `on_player_finished_playing()` should be triggered and detect that the autoplaylist is enabled.

This is probably the most simple implementation:

```py
self.config.auto_playlist = not self.config.auto_playlist
return Response("The autoplaylist is now " + ['disabled', 'enabled'][self.config.auto_playlist])